ConstructorGenerator/Model/Helper/MemberValidator.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using ConstructorGenerator.Attributes;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace ConstructorGenerator.Model.Helper;
+
+public class MemberValidator
+{
+    public bool ShouldGenerateConstructorParameterForMember(ISymbol memberSymbol, bool generateFullConstructorRequested)
+    {
+        if (WellKnownAttributes.ExcludeConstructorDependencyAttribute.IsDefined(memberSymbol)) 
+            return false; // this member is excluded
+
+        if (memberSymbol.IsImplicitlyDeclared)
+            return false; // funny compiler field
+
+        AttributeData? constructorDependencyAttributeData = WellKnownAttributes.ConstructorDependencyAttribute.GetAttributeData(memberSymbol);
+        if (!generateFullConstructorRequested && constructorDependencyAttributeData == null)
+        {
+            // GenerateFullConstructorAttribute does not exists for the class and ConstructorDependencyAttribute does not exists for this member
+            return false;
+        }
+        
+        if (constructorDependencyAttributeData == null && IsInitialized(memberSymbol))
+        {
+            return false;
+        }
+        
+        return memberSymbol switch
+        {
+            IPropertySymbol propertySymbol => ShouldGenerateConstructorForProperty(propertySymbol,
+                constructorDependencyAttributeData),
+            IFieldSymbol fieldSymbol => ShouldGenerateConstructorForField(fieldSymbol,
+                constructorDependencyAttributeData),
+            _ => false
+        };
+    }
+
+    private bool ShouldGenerateConstructorForField(IFieldSymbol fieldSymbol, AttributeData? explicitConstructorDependencyAttributeData)
+    {
+        return explicitConstructorDependencyAttributeData != null || fieldSymbol.IsReadOnly;
+    }
+
+    private bool ShouldGenerateConstructorForProperty(IPropertySymbol propertySymbol, AttributeData? explicitConstructorDependencyAttributeData)
+    {
+        if (!IsAutoProperty(propertySymbol))
+            return false; // non-auto properties should be ignored because there is a backing field.
+
+        // property should be not read-only or init-only OR explicit ConstructorDependencyAttribute is present
```

Comment:
The comment is the wrong way round. A ctor parameter is generated if the property is get or init only. Also, perhaps get rid of the double negative in the method name and usage.